src/routes/usersRoute.js

```diff
@@ -0,0 +1,59 @@
+import { Router } from "express";
+import short from "short-uuid";
+import users from "../data/users.js";
+const router = Router();
+
+router.get("/", (req, res) => {
+  res.status(200).json("Welcome To Home Page");
+});
+
+router.get("/users", (req, res) => {
+  if (users.length === 0) {
+    return res.status(404).json({ message: "Users not found" });
+  }
+  res.json(users);
+});
+
+router.get("/users/country=:country", (req, res) => {
+  const { country } = req.params;
+
+  const filteredUsers = users.filter(
+    (user) => user.country.toLowerCase() === country.toLowerCase()
+  );
+  if (!filteredUsers || filteredUsers.length === 0) {
+    return res.status(404).json({ message: `No Person found from ${country}` });
+  }
+  res.status(200).json(filteredUsers);
+});
```

Comment:
It's OK! However, The URL structure is designed as "/users/country=:country", which generally does not adhere to RESTful API design principles. A more RESTful approach would be to use query parameters for filtering by country, for example: /users?country=Canada. This keeps the URL cleaner and facilitates the addition of multiple query parameters.

This is the RESTful Design Principles.

**Query parameters are especially useful for:**

- **Filtering**: To narrow down the results based on specific criteria, such as /users?country=Canada.
- **Sorting**: To sort the results in a specific order, such as /users?sortBy=name&order=asc.
- **Searching**: To search for resources that match a query, such as /users?search=John.
- **Pagination**: To limit the results to a specific number per page and to navigate between pages, such as /users?page=2&limit=10.